cli/flox/src/commands/build.rs

```diff
@@ -166,6 +167,7 @@ impl Build {
         });
 
         check_stability_compatibility(&packages_to_build, nixpkgs_url_select.is_some())?;
+        check_git_repo_for_expression_builds(&packages_to_build, &base_dir)?;
```

Comment:
Do we also need to implement this for `flox publish`? In which case should it be a concern of the build provider?

Reply:
flox publish already checks for a git repo for publish reasons, checking again for builds seems a bit redundant.